<launch>
  <param name="/use_sim_time" value="false"/>
  <arg name="reso_x"              default="1280"/>
  <arg name="reso_y"              default="720"/>
  <arg name="fps"                 default="6"/>
  
  <arg name="serial_no"           default="839112060219"/>
  <arg name="json_file_path"      default=""/>
  <arg name="camera"              default="camera_basket"/>
  <arg name="tf_prefix"           default="$(arg camera)"/>

  <arg name="fisheye_width"       default="$(arg reso_x)"/>
  <arg name="fisheye_height"      default="$(arg reso_y)"/>
  <arg name="enable_fisheye"      default="true"/>

  <arg name="depth_width"         default="$(arg reso_x)"/>
  <arg name="depth_height"        default="$(arg reso_y)"/>
  <arg name="enable_depth"        default="true"/>

  <arg name="infra_width"        default="$(arg reso_x)"/>
  <arg name="infra_height"       default="$(arg reso_y)"/>
  <arg name="enable_infra1"       default="true"/>

  <arg name="enable_infra2"       default="true"/>

  <arg name="color_width"         default="$(arg reso_x)"/>
  <arg name="color_height"        default="$(arg reso_y)"/>
  <arg name="enable_color"        default="true"/>

  <arg name="fisheye_fps"         default="$(arg fps)"/>
  <arg name="depth_fps"           default="$(arg fps)"/>
  <arg name="infra_fps"          default="$(arg fps)"/>
  <arg name="color_fps"           default="$(arg fps)"/>
  <arg name="gyro_fps"            default="1000"/>
  <arg name="accel_fps"           default="1000"/>

  <arg name="enable_pointcloud"         default="true"/>
  <arg name="pointcloud_texture_stream" default="RS2_STREAM_COLOR"/>
  <arg name="pointcloud_texture_index"  default="0"/>

  <arg name="enable_sync"         default="false"/>
  <arg name="align_depth"         default="true"/>

  <arg name="filters"             default=""/>

  <group ns="$(arg camera)">
    <include file="$(find realsense2_camera)/launch/includes/nodelet.launch.xml">
      <arg name="tf_prefix"                value="$(arg tf_prefix)"/>
      <arg name="serial_no"                value="$(arg serial_no)"/>
      <arg name="json_file_path"           value="$(arg json_file_path)"/>

      <arg name="enable_pointcloud"        value="$(arg enable_pointcloud)"/>
      <arg name="pointcloud_texture_stream" value="$(arg pointcloud_texture_stream)"/>
      <arg name="pointcloud_texture_index"  value="$(arg pointcloud_texture_index)"/>
      <arg name="enable_sync"              value="$(arg enable_sync)"/>
      <arg name="align_depth"              value="$(arg align_depth)"/>

      <arg name="fisheye_width"            value="$(arg fisheye_width)"/>
      <arg name="fisheye_height"           value="$(arg fisheye_height)"/>
      <arg name="enable_fisheye"           value="$(arg enable_fisheye)"/>

      <arg name="depth_width"              value="$(arg depth_width)"/>
      <arg name="depth_height"             value="$(arg depth_height)"/>
      <arg name="enable_depth"             value="$(arg enable_depth)"/>

      <arg name="color_width"              value="$(arg color_width)"/>
      <arg name="color_height"             value="$(arg color_height)"/>
      <arg name="enable_color"             value="$(arg enable_color)"/>

      <arg name="infra_width"             value="$(arg infra_width)"/>
      <arg name="infra_height"            value="$(arg infra_height)"/>
      <arg name="enable_infra1"            value="$(arg enable_infra1)"/>

      <arg name="enable_infra2"            value="$(arg enable_infra2)"/>

      <arg name="fisheye_fps"              value="$(arg fisheye_fps)"/>
      <arg name="depth_fps"                value="$(arg depth_fps)"/>
      <arg name="infra_fps"               value="$(arg infra_fps)"/>
      <arg name="color_fps"                value="$(arg color_fps)"/>
      <arg name="gyro_fps"                 value="$(arg gyro_fps)"/>
      <arg name="accel_fps"                value="$(arg accel_fps)"/>

      <arg name="filters"                  value="$(arg filters)"/>
    </include>
  </group>
</launch>
